### DISCUSSION TITLE
Action not marked as failed if an error occurs

### DISCUSSION THREAD

**Discussion by Nef10** [BetaHuhn/repo-file-sync-action#95] [Ideas] [answered]
**🐞 Describe the bug**

The actions run is marked as success even though an error occurred.

**📚 To Reproduce**

Use an invalid token, or a token which does not have correct access.

**💡 Expected behavior**

The workflow run should be marked as failed.

**🖼️ Screenshots**

See
https://github.com/Nef10/swift-package-common/actions/runs/1178374879
https://github.com/Nef10/swift-package-common/actions/runs/1178388705

**⚙️ Environment**

`BetaHuhn/repo-file-sync-action@v1`

**Comment by BetaHuhn** [comment]
Actually, just remembered why it is handled that way:

When syncing files to multiple repositories the action goes through each one individually. If an error occurs while handling one, it just logs it and goes to the next repository instead of failing the whole action. The action only fails when an error occurs during startup/getting the inputs and when parsing the sync config.

I guess when it only syncs to one repository it makes sense to fail the action when an error occurs, else the current approach is better IMO as it will at least try to sync to the other repositories if one fails.

What do you think @Nef10?

**Comment by Nef10** [comment, marked as answer]
I think it is good that it continues to sync other repos, but should it be marked failed at the end? Like it saves a flag that was an error with some sync and just marks the actions as failed at the end.

A failed action makes it easier to spot errors, for example it triggers an emails (well, depending on your notification settings, but I think by default).

**Reply by BetaHuhn** [reply]
Yeah, makes sense. It should fail the action but still try to sync as many repositories as possible.

(Moved this to discussions as it isn't a bug)

**Reply by BetaHuhn** [reply]
I don't have much time right now to work on this myself but would review/merge a PR for this if you create one.

**Reply by BetaHuhn** [reply]
Added this with [v1.15.2](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.15.2).
